ctru-rs/src/applets/swkbd.rs

```diff
@@ -1,24 +1,36 @@
 //! Software Keyboard applet.
 //!
 //! This applet opens a virtual keyboard on the console's bottom screen which lets the user write UTF-16 valid text.
-// TODO: Implement remaining functionality (password mode, filter callbacks, etc.). Also improve "max text length" API. Improve `number of buttons` API when creating a new SoftwareKeyboard.
-// TODO: Split the Parental PIN lock operations into a different type.
 #![doc(alias = "keyboard")]
 
+use crate::services::{apt::Apt, gfx::Gfx};
+use ctru_sys::{self, SwkbdState};
+
 use bitflags::bitflags;
-use ctru_sys::{
-    self, swkbdInit, swkbdInputText, swkbdSetButton, swkbdSetFeatures, swkbdSetHintText,
-    swkbdSetInitialText, SwkbdState,
-};
 use libc;
+
+use std::borrow::Cow;
+use std::ffi::CStr;
 use std::fmt::Display;
 use std::iter::once;
 use std::str;
 
+type CallbackFunction = dyn Fn(&CStr) -> (CallbackResult, Option<Cow<CStr>>);
+
 /// Configuration structure to setup the Software Keyboard applet.
 #[doc(alias = "SwkbdState")]
+pub struct SoftwareKeyboard<'a> {
```

Comment:
nit: Add documentation to explain why there is a lifetime

Reply:
The lifetime requirement has been removed.